Library/Homebrew/dev-cmd/create.rb

```diff
@@ -69,6 +69,24 @@ def create_args
     end
   end
 
+  def url2name(url)
+    odebug "url: #{url}"
+    stem = Pathname.new(url).stem
+    odebug "stem: #{stem}"
+    # special cases first
+    name = if stem.start_with? "index.cgi"
+      # http://www.codesrc.com/gitweb/index.cgi?p=libzipper.git;a=summary
+      # https://github.com/Homebrew/brew/pull/9459
+      stem.rpartition("=").last
+    elsif url =~ %r{github\.com/\S+/(\S+)/(archive|releases)/}
```

Comment:
would be nicer to add these regex into `case url` in `formula_creator.rb`

Reply:
The complexity of the code in `fc.url` is not for the faint of the heart. Otherwise these regex would be there from the start. I need to move name parsing out of it first to ensure the test coverage.

Reply:
Happy to help you figure it out but I think it makes sense for this code to remain there rather than moving to `dev-cmd/create.rb` and add the tests for the `FormulaCreator` class instead.

Reply:
Moved the code into `formula_creator.rb`, but I am hesitant to make it more complicated by moving it into a class after decoupling it from there. Class coupling will makes both the code and the tests less readable, as it will assume internal states that do not exist here.

Reply:
Sorry, the goal is to minimise the amount of changes into the existing code, not minimise the amount of changes required to your code to get this PR merged. 

If you have issues with how the code is organised: changing that at the same time as fixing a bug is not a good combination.

Reply:
I see. I am going to split that PR then when time permits.